### DISCUSSION TITLE
msgmerge dreaming up translations for new messages?

### DISCUSSION THREAD

**Discussion by yohcop** [google/comprehensive-rust#919] [Q&A] [answered]
Hi,

I just sent a pull request, with only a clean `msgmerge --update` command.

However, I see a handful of translations that seems to be completely invented. I added comments to my PR, in order to try to understand what's going on, or what I may be doing wrong.

https://github.com/google/comprehensive-rust/pull/918

All I did, was 

```shell
$ MDBOOK_OUTPUT='{"xgettext": {"pot-file": "messages.pot"}}'   mdbook build -d po
$ msgmerge --update po/fr.po po/messages.pot 
```

And in the diffs, I see things like:

```po
#: src/SUMMARY.md:243
#, fuzzy
msgid "Exceptions"
msgstr "Fonctions"
```

Where does the "Fonctions" string come from? I can't even see it in the previous version of the file?

Thanks,

**Comment by mgeisler** [comment, marked as answer]
Hey @yohcop,

Great question, thanks for asking this! I wrote up a bunch about it [here](https://github.com/google/comprehensive-rust/pull/918#discussion_r1255799774) before seeing your question.

In short, `msgmerge` is trying hard to preserve existing translations — I don't know precisely how it does this and it seems like it can be pretty creative in how it does this :smile: When it guesses, it marks the entry with `fuzzy`, which indicates that a human should proof-read it. We don't use fuzzy entries at all when rendering the book, so it's important to clear out those fields later (probably a second PR, though I would combine it when the course is fully translated).

(cc @henrif75: more translation questions)
